src/psyclone/psyir/frontend/fparser2.py

```diff
@@ -2218,6 +2219,39 @@ def process_declarations(self, parent, nodes, arg_list,
                     # Restore the fparser2 parse tree
                     node.children[2].items = tuple(orig_children)
 
+            elif isinstance(node, Fortran2003.Common_Stmt):
+
+                # If the first common object has a name, use it
+                if node.children[0][0][0]:
+                    name = node.children[0][0][0]
+                else:
+                    name = "unnamed"
+
+                # Place the declaration statement into a UnknownFortranType
+                # (for now we just want to reproduce it). The name of the
+                # codeblock is not in the same namespace as the variable
+                # symbols names, so we add a psyclone prefix to
+                # differentiate them.
+                parent.symbol_table.new_symbol(
+                    f"_PSYCLONE_COMMONBLOCK_{name}",
+                    symbol_type=DataSymbol,
+                    datatype=UnknownFortranType(str(node)))
+
+                # Get the names of the symbols accessed with the commonblock,
+                # they are already defined in the symbol table but they must
+                # now have a common-block interface.
+                try:
+                    for cb_object in node.children[0]:
+                        for symbol_name in cb_object[1].items:
+                            sym = parent.symbol_table.lookup(str(symbol_name))
+                            sym.interface = CommonBlockInterface()
```

Comment:
UPDATE: I can see now that this would be problematic so ignore me.

Reply:
I think they could and I was thinking of adding them during the first version of this PR. But since there can be some complex cases I think is better to wait until we have a use case for it. At the moment we just need basic support to avoid the routines ending up as CodeBlocks.